File: WorkManagerJava/app/src/main/java/com/websarva/wings/android/workmanagerjava/CountUpWorker.java
package com.websarva.wings.android.workmanagerjava;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class CountUpWorker extends Worker {
	public CountUpWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
		super(context, workerParams);
	}

	@NonNull
	@Override
	public Result doWork() {
		for(int i = 1; i <= 25; i++) {
			Log.i("CountUpWorker", "現在のカウント: " + i);
			try {
				Thread.sleep(1000);
			} catch (InterruptedException e) {
				return Result.failure();
			}
		}
		return Result.success();
	}
}

File: WorkManagerKotlin/app/src/main/java/com/websarva/wings/android/workmanagerkotlin/CountUpWorker.kt
package com.websarva.wings.android.workmanagerkotlin

import android.content.Context
import android.util.Log
import androidx.work.Worker
import androidx.work.WorkerParameters

class CountUpWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {
	override fun doWork(): Result {
		for(i in 1..25) {
			Log.i("CountUpWorker", "現在のカウント: ${i}")
			Thread.sleep(1000)
		}
		return Result.success()
	}
}
